src/Api/API.php

```diff
@@ -1276,6 +1276,12 @@ protected function getItems($itemtype, $params = [], &$totalcount = 0)
 
                 $found[] = $data;
             }
+        } else {
+            $this->returnError(
+                'Error while querying database.' . $this->returnError($DB->error()),
+                400,
+                "ERROR_DATABASE_QUERY",
+            );
```

Comment:
Exposing the database error messages to the API is not a good idea. Indeed, it will probably not help the API client, as it will, most of the time, the result of a server side bug. Also, it may expose data that should not be exposed, and maybe also help an attacker to understand that there is a potential SQL injection to exploit.

IMHO, we should not even says that the error is related to the database.

```suggestion
            $this->returnError(
                'An error occurred during the items search.',
                400,
                "ERROR_UNKNOWN",
            );
```

Reply:
The message shown in the new API when the iterator result is invalid is:
'An internal error occured while trying to fetch the data.'
In debug mode, it appends 'For more information, check the GLPI logs.' to that message as well.

This should probably be used in the legacy API too.